### PR TITLE
Adding tasks to publish test results to AzurePipelines

### PR DESCRIPTION
**Pull request by vagisha-nidhi**
**What kind of change does this PR introduce?**

This PR is related to the issue #589 
The PR helps to publish test results for reporting in Azure Pipelines.

**Summary**

Azure Pipelines provides excellent reporting capabilities that can be leveraged by this PR.
An example report is shown in the image below.
image
A screenshot for code coverage : 
image

**Does this PR introduce a breaking change?**
No

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -10,16 +10,16 @@
...
     "start": "npm run test:watch",
     "lint": "eslint --ignore-path .gitignore .",
-    "test": "nyc --require @babel/register _mocha -- test/tests/index.js",
+    "test": "nyc --require @babel/register _mocha --reporter=mocha-multi-reporters --reporter-options configFile=./test/mochareporters.json -- test/tests/index.js",
     "test:watch": "nodemon -q --ignore test/.tmp/ --ignore test/artifacts/ --ignore coverage/ --exec \"npm run test\" --"
   },
   "config": {
     "commitizen": {
-      "path": "./node_modules/cz-conventional-changelog"
+      "path": "./node_modules/cz-conventional-changelog-default-export"
```
Comment (jimthedev) [inline package.json]: Hey! This looks pretty good but I was wondering if this line is intentional?

**Comment by jimthedev** [comment]
@vagisha-nidhi did you see my feedback above?

File: package.json (hunk 1 above)
Comment (vagisha-nidhi) [reply package.json]: @jimthedev Hey. Sorry I missed this conversation. This change wasn't intentional and not related to the working I guess.

Comment (jimthedev) [reply package.json]: If you can update it then I can merge.

**Pull request merged**